File: Greedy/19941번 - 햄버거/19941.cpp
#include <iostream>

using namespace std;

#define LIM 20001

int n, k;
string str;

void input() {
    cin >> n >> k;
    cin >> str;
}

void solve() {
    int ans = 0;
    
    for(int i = 0; i < n; i++) {
        if(str[i] != 'P') continue;

        for(int j = i - k; j <= i + k; j++) {
            if((0 <= j && j < n) && str[j] == 'H') {
                str[j] = '-';
                ans++;
                break;
            }
        }
    }
    
    cout << ans << '\n';
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    input();
    solve();
    
    return 0;
}

File: Greedy/19941번 - 햄버거/19941.py
import sys

input = sys.stdin.readline

n, k = map(int, input().split())
table = list(input().rstrip())

def solve():
    ans = 0
    
    for i in range(n):
        if table[i] != 'P':
            continue
        
        for j in range(max(i - k, 0), min(i + k + 1, n)):
            if table[j] == 'H':
                table[j] = '-'
                ans += 1
                break
    
    print(ans)

if __name__ == '__main__':
    solve()